### ISSUE TITLE
GPU内存泄露？

### ISSUE THREAD

**Issue by jeffzhengye** [PaddlePaddle/PaddleSpeech#1235]
WSL + paddlepaddle2.2.1 + 3090 24G
fine-tune conformer_wenetspeech-zh-16k 这个与训练模型，batch_size=64 正常的时候使用gpu内存10G左右，但是时不时内存会飙升到20多G然后又降到10G左右，但有时候也会内存耗尽崩溃。是不是程序有内存泄漏啊？
配置文件如下：

# network architecture
model:
    # encoder related
    encoder: conformer
    blank_id: 0
    encoder_conf:
        output_size: 512    # dimension of attention
        attention_heads: 8
        linear_units: 2048  # the number of units of position-wise feed forward
        num_blocks: 12      # the number of encoder blocks
        dropout_rate: 0.1
        positional_dropout_rate: 0.1
        attention_dropout_rate: 0.0
        input_layer: conv2d # encoder input type, you can chose conv2d, conv2d6 and conv2d8
        normalize_before: True
        use_cnn_module: True
        cnn_module_kernel: 15
        cnn_module_norm: layer_norm
        activation_type: swish
        pos_enc_layer_type: rel_pos
        selfattention_layer_type: rel_selfattn

    # decoder related
    decoder: transformer
    decoder_conf:
        attention_heads: 8
        linear_units: 2048
        num_blocks: 6
        dropout_rate: 0.1
        positional_dropout_rate: 0.1
        self_attention_dropout_rate: 0.0
        src_attention_dropout_rate: 0.0

    # hybrid CTC/attention
    model_conf:
        ctc_weight: 0.3
        ctc_dropoutrate: 0.0
        ctc_grad_norm_type: null
        lsm_weight: 0.1     # label smoothing option
        length_normalized_loss: false

# https://yaml.org/type/float.html
data:
  train_manifest: data/manifest.train
  dev_manifest: data/manifest.test
  test_manifest: data/manifest.test
  min_input_len: 0.0
  max_input_len: 12.0
  min_output_len: 1.0
  max_output_len: 400.0
  min_output_input_ratio: 0.05
  max_output_input_ratio: 20.0

collator:
  vocab_filepath: data/vocab.txt 
  unit_type: 'char'
  spm_model_prefix: ''
#  augmentation_config: conf/preprocess.yaml
  augmentation_config: conf/augmentation.json
  batch_size: 64
  raw_wav: True  # use raw_wav or kaldi feature
  spectrum_type: fbank #linear, mfcc, fbank
  feat_dim: 80
  delta_delta: False
  dither: 1.0
  target_sample_rate: 16000
  max_freq: None
  n_fft: None
  stride_ms: 10.0
  window_ms: 25.0
  use_dB_normalization: True 
  target_dB: -20
  random_seed: 0
  keep_transcription_text: False
  sortagrad: True 
  shuffle_method: batch_shuffle
  num_workers: 1

training:
  n_epoch: 40
  accum_grad: 1
  global_grad_clip: 5.0
  log_interval: 400
  checkpoint:
    kbest_n: 50
    latest_n: 5
  optim: adam
  optim_conf:
    lr: 0.0005
    weight_decay: 1e-6
  scheduler: warmuplr     # pytorch v1.1.0+ required
  scheduler_conf:
    warmup_steps: 5000
    lr_decay: 1.0

decoding:
  batch_size: 64
  error_rate_type: cer
#  decoding_method: ctc_greedy_search # 'attention', 'ctc_greedy_search', 'ctc_prefix_beam_search', 'attention_rescoring'
  decoding_method: ctc_beam_search # "ctc_greedy", "ctc_beam_search", ctc.py中只支持这两个
  lang_model_path: data/lm/common_crawl_00.prune01111.trie.klm
  alpha: 2.5
  beta: 0.3
  beam_size: 5
  cutoff_prob: 1.0
  cutoff_top_n: 0
  num_proc_bsearch: 8
  ctc_weight: 0.5 # ctc weight for attention rescoring decode mode.
  decoding_chunk_size: -1 # decoding chunk size. Defaults to -1.
      # <0: for decoding, use full chunk.
      # >0: for decoding, use fixed chunk size as set.
      # 0: used for training, it's prohibited here. 
  num_decoding_left_chunks: -1  # number of left chunks for decoding. Defaults to -1.
  simulate_streaming: False  # simulate streaming inference. Defaults to False.

**Comment by Jackwaterveg** [comment]
因为每个batch的数据长度是不相同的，所以需要的显存大小会根据batch的数据长度发生变化。建议调小batchsize进行训练。
